### DISCUSSION TITLE
Custom dynamic/effect

### DISCUSSION THREAD

**Discussion by steffensylvest** [sbidy/wiz_light#107] [General]
Do anybody know if it is possible to create customized "dynamics" (isn't that what WiZ calls it?). I want to create 2 effects: 

1. I want to create a 5 sec. transition from lowest brightness to full brightness, which I will use to do a "smoothless" dim-up/dim-down feature when I click a physical button (called through Home Assistant).

2. Sunrise where I can determine how long it should take (rather than the 30 minutes default)

I realise the limitations of the transitions (as mentioned in a recent status update), but perhaps the work-around could be to create custom effects/dynamics - if it is possible at all?

Thanks!

/Steffen

**Comment by sbidy** [comment]
It is on my backlog but there are so much other issues to fix fist.

You can maybe create a workaround if you add the bulbs in the App to different "rooms" and set the transition to ~2sec. and trigger it via HASS. As I mentioned here https://github.com/sbidy/wiz_light#what-is-declined-or-rejected the transition can only be set via app. And HASS doesn't support these feature in the default light integration.

Another solution for the Sunrise event is maybe a HASS script. You can pull up the brightness according to the Sun state etc. Maybe google will find some examples.

But thank you for the feedback!

**Comment by ChrisLizon** [comment]
For your sunrise ideas, it may be worthwhile looking into the [Flux](https://www.home-assistant.io/integrations/flux/), [Circadian_lighting](https://github.com/claytonjn/hass-circadian_lighting), or [Adaptive Lighting](https://github.com/basnijholt/adaptive-lighting) components. One of them may be able to be leveraged to do what you would like.

**Comment by steffensylvest** [comment]
Thanks for suggestions.. Will try to work around it for now :-)
